projects/packages/forms/src/contact-form/js/accessible-form.js

```diff
@@ -269,6 +269,7 @@ const isDateFieldValid = input => {
 	if ( value && format && typeof $ !== 'undefined' ) {
 		try {
 			$.datepicker.parseDate( format, value );
+			input.setCustomValidity( '' );
```

Comment:
The below `setCustomValidity` call would result in following invalidations of the field ([customError](https://developer.mozilla.org/en-US/docs/Web/API/ValidityState/customError)) and needs to be reset.